common/unsafe/src/main/java/org/apache/spark/unsafe/memory/HeapMemoryAllocator.java

```diff
@@ -47,23 +47,29 @@ private boolean shouldPool(long size) {
 
   @Override
   public MemoryBlock allocate(long size) throws OutOfMemoryError {
-    if (shouldPool(size)) {
+    int arraySize = (int)((size + 7) / 8);
```

Comment:
You might be able to use `ByteAraryMethods.roundNumberOfBytesToNearestWord` for this, which we'e done for similar rounding elsewhere. Makes it a bit easier to spot what's happening.

Reply:
But the type of input parameter for `roundNumberOfBytesToNearestWord` is `int`

Reply:
Maybe we should make the method to tackle long values.